internal/vault/responses.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import "github.com/hashicorp/vault/api"
+
+var (
+	_ Response = (*defaultResponse)(nil)
+	_ Response = (*kvV2Response)(nil)
+)
+
+type Response interface {
+	Secret() *api.Secret
+	Data() map[string]interface{}
```

Comment:
Is there anything preventing this from returning `map[string]any` instead?

Reply:
Nope, should be fine to do so.

Reply:
Updated with a74d5224b88c8929d26ee7380886c267a29d10aa